File: java-samples/Advanced/snakesandladders-cordapp/workflows/src/main/java/net/corda/samples/snl/flows/StartGameFlow.java
package net.corda.samples.snl.flows;

import co.paralleluniverse.fibers.Suspendable;
import net.corda.core.flows.FlowException;
import net.corda.core.flows.FlowLogic;
import net.corda.core.flows.StartableByRPC;
import net.corda.core.transactions.SignedTransaction;

@StartableByRPC
public class StartGameFlow extends FlowLogic<String> {

    private String player1;
    private String player2;

    public StartGameFlow(String player1, String player2) {
        this.player1 = player1;
        this.player2 = player2;
    }

    @Override
    @Suspendable
    public String call() throws FlowException {
        subFlow(new CreateBoardConfig.Initiator(player1, player2));
        return subFlow(new CreateGameFlow.Initiator(player1, player2));
    }
}

File: kotlin-samples/Advanced/snakesandladders-cordapp/workflows/src/main/kotlin/net/corda/samples/snl/flows/StartGameFlow.kt
package net.corda.samples.snl.flows

import co.paralleluniverse.fibers.Suspendable
import net.corda.core.flows.FlowException
import net.corda.core.flows.FlowLogic
import net.corda.core.flows.StartableByRPC

@StartableByRPC
class StartGameFlow(private val player1: String, private val player2: String) : FlowLogic<String>() {
    @Suspendable
    @Throws(FlowException::class)
    override fun call(): String {
        subFlow(CreateBoardConfig.Initiator(player1, player2))
        return subFlow(CreateGameFlow.Initiator(player1, player2))
    }
}